src/provider/types.ts

```diff
@@ -0,0 +1,213 @@
+import { EventEmitter } from 'events'
+import { Duplex } from 'stream'
+import {
+    JSONRPCRequestPayload as RequestPayload,
+    JSONRPCResponsePayload as ResponsePayload,
+    JSONRPCErrorCallback,
+    Transaction,
+} from 'ethereum-protocol'
+import Web3ProviderEngine from 'web3-provider-engine'
+import { IConnector as WalletConnector } from '@walletconnect/types'
+
+interface AnyObject {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [K: string]: any
+}
+
+export type JSONRPCResponsePayload = Partial<ResponsePayload> &
+    Pick<ResponsePayload, 'result'> & {
+        error?: FormattedError
+    }
+export type JSONRPCRequestPayload = Partial<RequestPayload> & Pick<RequestPayload, 'method'>
+
+export { WalletConnector }
+
+type RpcMiddleware = (
+    req: JSONRPCRequestPayload,
+    res: JSONRPCResponsePayload,
+    next: (cb?: Error | Function) => void,
+    end: (err?: Error) => void,
+) => void
+
+interface RpcEngine extends EventEmitter {
+    handle(req: JSONRPCRequestPayload, callback: JSONRPCErrorCallback): void
+    push(middleware: RpcMiddleware): void
+}
+
+interface ObservableStore<T = AnyObject> extends EventEmitter {
+    getState(): T
+    pushState(newState: T): void
+    updateState(partialState: Partial<T>): void
+    subscribe(handler: Parameters<EventEmitter['on']>[1]): void
+    unsubscribe(handler: Parameters<EventEmitter['on']>[1]): void
```

Comment:
Where is `Parameters` coming from? I see no import or declaration.

Reply:
It comes with TS, same as `Pick | Partial| ...`. `Ctrl+LM` on it in VSCode to see every type ever